wordpress/wp-content/themes/cds-default/inc/template-functions.php

```diff
@@ -244,6 +244,55 @@ function language_switcher_output($languages)
 
     return $langs;
 }
+/*
+    TODO: Replace this hack function with a proper solution.
+*/
+/*
+    * This function is a total hack to fix a WPML bug.
+    * The `convert_url` function used by WPML does not work properly
+    * when the page is a `category` page. This function is a hack to
+    * fix that issue.
+    *
+    * WARNING: This function assumes only two languages: EN and FR.
+    *
+    * WARNING: `$category_path_name` is hardcoded, and will break if the category path
+    * is changed within the Wordpress dashboard for any of the sites.
+    *
+    * @param string $url - the original, incorrect, `translated_url` provided by the
+    *                      get_ls_languages() function, found within the
+    *                      `wpml_active_languages` apply_filter hook.
+    * @param string $lang - the language code for the target language
+    * @return string - the correct url for the target language
+    *                   the url will have the root path for the target language,
+    *                   and the category slug for the current category, in the
+    *                   current (non-target) language. When Wordpress detects the language
+    *                   in the category slug, it will redirect to the correct category slug
+    *                   in the target language, based on the root path.
+    *
+    *                   eg: On the EN page for a category called "test-en", the incorrect url
+    *                   for the language switcher will be:
+    *                       ~/category/test-en
+    *
+    *                   This function will convert that url to:
+    *                       ~/fr/category/test-en
+    *
+    *                   Wordpress will then redirect to the correct url for the target language:
+    *                       ~/fr/category/test-fr
+*/
+function convert_url($url, $lang): string
+{
+    $category_path_name = "category";
+    $return_url = $url;
+    $parsed_url = parse_url($return_url);
+    if (str_contains($parsed_url['path'], "/$category_path_name/")) { // only modify the url if it is a category page
```

Comment:
Haven't tested this PR yet but might suggest getting rid of the else

```php
if (str_contains($parsed_url['path'], "/$category_path_name/")) { // only modify the url if it is a category page
    if (str_starts_with($parsed_url['path'], "/$category_path_name/")) { // currently on the EN path
        $return_url = str_replace("/$category_path_name/", "/$lang/$category_path_name/", $return_url);
    } 
    
    // otherwise it must be on the FR path
    return $return_url = str_replace("/fr/", "/", $return_url);
        
}
```